ts/src/openapi/src/models/IssuedProductServicingData.ts

```diff
@@ -53,7 +53,7 @@ import {
  * The data needed to service this specific type of issued product
  * @export
  */
-export type IssuedProductServicingData = IntegratedMcaServicingData | InvoiceFinancingServicingData | LineOfCreditServicingData | McaServicingData | TermLoanServicingData;
+export type IssuedProductServicingData = { productType: 'INTEGRATED_MCA' } & IntegratedMcaServicingData | { productType: 'INVOICE_FINANCING' } & InvoiceFinancingServicingData | { productType: 'LINE_OF_CREDIT' } & LineOfCreditServicingData | { productType: 'MCA' } & McaServicingData | { productType: 'TERM_LOAN' } & TermLoanServicingData;
```

Comment:
why does this union look so strange? vs. this one looks more normal
```
export type InvoicePaymentPlanRepaymentWindow = FixedDateInvoiceRepaymentWindow | FixedDurationInvoiceRepaymentWindow | MultipleDurationInvoiceRepaymentWindow;
```

they both probably work, but `InvoicePaymentPlanRepaymentWindow` looks more clean to me. and i'm wondering why there's a difference

Reply:
As discuss in Slack, 
if we want the auto unwrap works like what we do in repayment windows, we need to redefine the enum per usage. But the description is missing in the documentation. Somehow it still works for repayment window, but just not work for product type........ Parent and child annotations are exactly the same.....

I would prefer doing it with the current approach, using enum, enum name, and discriminator. Documentation looks good and one enum per type. The union type will look bad, but don't think people will use it a lot comparing to documentation and the enums.

<img width="1419" alt="Screenshot 2024-08-20 at 13 35 31" src="https://github.com/user-attachments/assets/9b1a4419-0dcf-4514-ac57-9e6f1793a140">
<img width="424" alt="Screenshot 2024-08-20 at 14 56 28" src="https://github.com/user-attachments/assets/eacf4e56-e2d4-4650-bdec-05c313543a00">